bot/exts/moderation/infraction/_scheduler.py

```diff
@@ -48,11 +48,32 @@ async def reschedule_infractions(self, supported_infractions: t.Container[str])
 
         infractions = await self.bot.api_client.get(
             'bot/infractions',
-            params={'active': 'true'}
+            params={
+                'active': 'true',
+                'ordering': 'expires_at',
+                'permanent': 'false',
+                'types': ','.join(supported_infractions),
+            },
         )
-        for infraction in infractions:
-            if infraction["expires_at"] is not None and infraction["type"] in supported_infractions:
-                self.schedule_expiration(infraction)
+
+        to_schedule = [i for i in infractions if not i['id'] in self.scheduler]
```

Comment:
Prefer `x not in y` over `not x in y`
```suggestion
        to_schedule = [i for i in infractions if i['id'] not in self.scheduler]
```